components/common/ListTable/ListTable.tsx

```diff
@@ -0,0 +1,128 @@
+
+  if (mdScreen)
+    return (
+      <Box style={{ justifyContent: 'flex-end', display: 'flex' }}>
+        <Button sx={IconButtonStyles}>{t('initiate-quote')}</Button>
+        <Button sx={IconButtonStyles}>{t('add-to-cart')}</Button>
+        <IconButton sx={IconButtonStyles}>
+          <Edit />
+        </IconButton>
+        <IconButton sx={IconButtonStyles}>
+          <FolderCopySharp />
+        </IconButton>
+        <IconButton sx={IconButtonStyles}>
+          <Delete />
+        </IconButton>
+      </Box>
+    )
+  return (
+    <IconButton style={{ padding: '0px' }}>
+      <MoreVert />
+    </IconButton>
+  )
+}
+
+const ListTable = (props: any) => {
+  const { rows } = props
+  const { t } = useTranslation('common')
+  const theme = useTheme()
+  const mdScreen = useMediaQuery(theme.breakpoints.up('md'))
+  return (
+    <TableContainer>
+      <Table style={{ tableLayout: 'fixed' }}>
+        <TableHead>
+          <TableRow style={{ backgroundColor: '#f7f7f7', padding: '10px 0' }}>
+            <TableCell style={{ padding: '10px 10px', width: mdScreen ? '25%' : '50%' }}>
+              {t('list-name')}
+            </TableCell>
+            <TableCell style={{ padding: '10px 10px', width: mdScreen ? '15%' : '30%' }}>
+              {t('date-created')}
+            </TableCell>
+            {mdScreen && (
+              <TableCell style={{ padding: '10px 10px', width: '20%' }}>
+                {t('created-by')}
+              </TableCell>
+            )}
+            <TableCell
+              style={{ padding: '10px 10px', width: mdScreen ? '25%' : '10%' }}
+            ></TableCell>
+          </TableRow>
+        </TableHead>
+        <TableBody>
+          {rows.map((item: any) => {
+            return (
+              <TableRow key={item.id}>
+                <TableCell style={{ ...tableCellStyles, width: mdScreen ? '25%' : '50%' }}>
+                  {mdScreen ? (
+                    item.name
+                  ) : (
+                    <Box>
+                      {item.name}
+                      <br />
+                      <p style={{ margin: '5px 0', color: '#cdcdcd' }}>{item.createBy}</p>
+                    </Box>
+                  )}
+                </TableCell>
+                <TableCell style={{ ...tableCellStyles, width: mdScreen ? '15%' : '30%' }}>
+                  {item.createDate}
+                </TableCell>
+                {mdScreen ? (
+                  <TableCell style={{ ...tableCellStyles, width: '20%' }}>
+                    {item.createBy}
+                  </TableCell>
+                ) : (
+                  <></>
+                )}
+                <TableCell style={{ ...tableCellStyles, width: mdScreen ? '25%' : '10%' }}>
+                  <ListItemOptions />
+                </TableCell>
+              </TableRow>
+            )
+          })}
+        </TableBody>
+      </Table>
+    </TableContainer>
+  )
+}
+
+export default ListTable
```

Comment:
I think making truly common `<Table/>` component between Users and WishtList is bit challenging. Better to move this component under my-account/List